bxin/ptg2hex/hex_diagnostics.ipynb

```diff
@@ -0,0 +1,607 @@
+    "    #according to XML, units are micron and degree\n",
+    "    await hex.cmd_offset.set_start(x=0,y=0,z=100*step, u=0,v=0,w=0,sync=True)\n",
+    "    await asyncio.sleep(0.1)\n",
+    "    state = await hex.evt_inPosition.aget(timeout=10)\n",
+    "    print(\"hex in position?\",state.inPosition, pd.to_datetime(state.private_sndStamp, unit='s'))\n",
+    "    hex.evt_inPosition.flush()\n",
+    "    state = await hex.evt_inPosition.next(flush=False, timeout=10)\n",
+    "    print('hex in position?', state.inPosition, pd.to_datetime(state.private_sndStamp, unit='s'))\n",
+    "    pos = await hex.tel_application.aget(timeout=10.)\n",
+    "    for i in range(6):\n",
+    "        print('%10.2f '%pos.position[i], end= ' ')\n",
+    "    print('\\n')\n",
+    "    #await asyncio.sleep(5)\n",
+    "    end = Time(datetime.now())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start = end - timedelta(seconds=30)\n",
+    "df = await client.select_time_series('lsst.sal.MTHexapod.application', '*', start.tai, end.tai, csc_index)\n",
+    "idx=df.MTHexapodID==1\n",
+    "df = df[idx]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots(figsize=(19,3))\n",
+    "plt.plot(df.position2)\n",
+    "plt.grid()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### When the LUT is enabled"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await hex.cmd_setCompensationMode.set_start(enable=1, timeout=10)\n",
+    "lutMode = await hex.evt_compensationMode.aget(timeout=10)\n",
+    "print(\"compsensation mode enabled?\",lutMode.enabled)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos = await hex.tel_application.aget(timeout=10.)\n",
+    "print('hex position')\n",
+    "for i in range(6):\n",
+    "    print('%10.2f '%pos.position[i], end= ' ')\n",
+    "print('\\n')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "posC = await hex.evt_compensatedPosition.aget(timeout=10.)\n",
+    "print('Compensated position')\n",
+    "for i in 'xyzuvw':\n",
+    "    print('%10.2f '%getattr(posC, i), end= ' ')\n",
+    "print('\\n')\n",
+    "posU = await hex.evt_uncompensatedPosition.aget(timeout=10.)\n",
+    "print('Uncompensated position')\n",
+    "for i in 'xyzuvw':\n",
+    "    print('%10.2f '%getattr(posU, i), end= ' ')\n",
+    "print('\\n')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "h = await mount.evt_heartbeat.next(flush=True, timeout=5)\n",
+    "print(h.heartbeat)"
```

Comment:
Why? The heartbeat sample doesn't contain anything useful.

Also i suggest flush=False. That way you are testing if the CSC is alive.